src/frontend/src/views/Upload/components/Metadata/components/Table/components/Row/components/LocationField/index.tsx

```diff
@@ -1,75 +1,126 @@
-import React from "react";
+import { escapeRegExp } from "lodash/fp";
+import React, { useEffect, useReducer, useState } from "react";
 import { Metadata } from "src/views/Upload/components/common/types";
 import ApplyToAllColumn from "../common/ApplyToAllColumn";
-import { COUNTIES } from "./COUNTIES";
-import { MenuSubtext, StyledTextField } from "./style";
-
-const SELECT_PROPS = {
-  displayEmpty: true,
-  renderValue,
-};
+import { StyledDiv } from "./style";
 
 interface Props {
   fieldKey: keyof Metadata;
+  accessoryKey: keyof Metadata;
   formik: FormikContextType<Metadata>;
   applyToAllColumn: (fieldKey: keyof Metadata, value: unknown) => void;
   isFirstRow: boolean;
+  locationOptions: GisaidLocationOption[];
+}
+
+interface LocationSearchState {
+  searching?: boolean;
+  results: GisaidLocationOption[];
+}
+
+function locationSearchReducer(
+  state: LocationSearchState,
+  action: LocationSearchState | Partial<LocationSearchState>
+): LocationSearchState {
+  return { ...state, ...action };
 }
 
 export default function LocationField({
   fieldKey,
+  accessoryKey,
   formik,
   applyToAllColumn,
   isFirstRow,
+  locationOptions,
 }: Props): JSX.Element {
-  const { handleChange, handleBlur, values, touched, errors } = formik;
+  const { handleBlur, setFieldValue, values, touched, errors } = formik;
 
   const errorMessage = touched[fieldKey] && errors[fieldKey];
 
-  const value = values[fieldKey] || "";
+  const value = values[fieldKey] || null;
 
-  return (
-    <StyledTextField
-      select
-      SelectProps={SELECT_PROPS}
-      name={fieldKey}
-      value={value}
-      margin="dense"
-      variant="outlined"
-      onChange={handleChange}
-      onBlur={handleBlur}
-      error={Boolean(errorMessage)}
-      helperText={
-        errorMessage ||
-        (isFirstRow && value && (
-          <ApplyToAllColumn
-            fieldKey={fieldKey}
-            value={value}
-            handleClick={applyToAllColumn}
-          />
-        ))
+  const [selectedLocation, setLocation] = useState<
+    GisaidLocationOption | undefined
+  >();
+  const [state, dispatch] = useReducer(locationSearchReducer, {
+    results: [],
+    searching: false,
+  });
+
+  useEffect(() => {
+    const locationForID = locationOptions.find(
+      (location) => location.id == value
+    );
+    setLocation(locationForID);
+  }, [locationOptions, value]);
+
+  const searcher = async (query: string): Promise<void> => {
+    if (query.length < 2) {
+      dispatch({ results: [] });
+      return;
+    }
+    dispatch({ searching: true });
+
+    const regex = new RegExp(escapeRegExp(query), "i");
+    const filteredLocationOptions = locationOptions.filter((option) =>
+      regex.test(option.name)
+    );
+    // alphabetical sort
+    // this ensures partial locations (i.e. region, country and divison
+    // but no location) end up on top.
+    const sortedLocationOptions = filteredLocationOptions.sort((a, b) => {
```

Comment:
I think this is the default sort behavior (ie, you can just say array.sort() instead of defining the whole thing here).